### DISCUSSION TITLE
[const char*]Support of Chinese charaters?

### DISCUSSION THREAD

**Discussion by huangweiwu** [wjakob/nanobind#295] [Q&A] [answered]
I detect the encoding from python string as below and the result is utf-8.
But the garbled characters problem still exists.

```
// python
s = "测试"
print(chardet.detect(s.encode())['encoding'])
m.test_func(s)

// C++
void test_func(const char* c) {};
m.def("test_func", "test_func");
```

**Comment by huangweiwu** [comment, marked as answer]
I've solved the problem.

**Reply by wjakob** [reply]
Could you explain what the issue was? (Perhaps for the benefit of future Chinese users of nanobind?)

**Reply by huangweiwu** [reply]
My C++ project use gbk and I use PyUnicode_AsGBKAndSize instead of PyUnicode_AsUTF8AndSize.
